sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2843,6 +2843,21 @@ object SQLConf {
       // show full stacktrace in tests but hide in production by default.
       .createWithDefault(Utils.isTesting)
 
+  val SPARK_ENRICH_ERROR_ENABLED =
+    buildConf("spark.sql.spark.enrichError.enabled")
+      .doc("When true, it enriches errors with full exception messages on the client side.")
+      .version("3.5.0")
```

Comment:
Would have to be `4.0.0` - I think it's too late for 3.5.0.

Reply:
and actually I think these configurations have to placed under `org.apache.spark.sql.connect.config.Connect`

Reply:
`org.apache.spark.sql.connect.config.Connect` cannot be configured by `spark.conf.set(...)` if I understand correctly?

Reply:
> Would have to be `4.0.0` - I think it's too late for 3.5.0.

Can `3.5.0` have this pr with the feature disabled by default?

Reply:
@heyihong if you want to set the conf without getting yelled at by SQLConf, use `onCreate(SQLConf.register)` when you are building the conf.

@grundprinzip that would make it a dynamic flag right?

Reply:
@heyihong 3.5.0 is too late.